libraries/AC_AutoTune/AC_AutoTune_Heli.h

```diff
@@ -0,0 +1,149 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  support for autotune of helicopters
+ */
+
+#pragma once
+
+#include "AC_AutoTune.h"
+
+class AC_AutoTune_Heli : public AC_AutoTune
+{
+public:
+    // constructor
+    AC_AutoTune_Heli();
+
+    // save gained, called on disarm
+    void save_tuning_gains() override;
+
+    // var_info for holding Parameter information
+    static const struct AP_Param::GroupInfo var_info[];
+
+protected:
+
+    void load_test_gains() override;
+
+    // get intra test rate I gain for the specified axis
+    float get_intra_test_ri(AxisType test_axis) override;
+
+    // get tuned rate I gain for the specified axis
+    float get_tuned_ri(AxisType test_axis) override;
+
+    // get tuned yaw rate d gain
+    float get_tuned_yaw_rd() override { return tune_yaw_rd; }
+
+    void test_init() override;
+    void test_run(AxisType test_axis, const float dir_sign) override;
+
+    // update gains for the rate p up tune type
+    void updating_rate_p_up_all(AxisType test_axis) override;
+
+    // update gains for the rate p down tune type
+    void updating_rate_p_down_all(AxisType test_axis) override {};
```

Comment:
It looks like updating_rate_p_down_all() is not implemented or called from anywhere?

Reply:
no.  its removed.